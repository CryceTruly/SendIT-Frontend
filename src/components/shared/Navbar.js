import React, { Component, Fragment } from 'react';
import {
  Collapse,
  Navbar,
  NavbarToggler,
  NavbarBrand,
  Nav,
  NavItem,
  NavLink,
  Container
} from 'reactstrap';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import {Link} from 'react-router-dom';
class Navigationbar extends Component {


  static propTypes = {
    auth: PropTypes.object.isRequired
  };




  render() {
    const { isAuthenticated, user } = this.props.auth;

    const authLinks = (
      <Fragment>
        <NavItem>
          <span className='navbar-text mr-3'>
            <strong>{user ? `Welcome ${user.name}` : ''}</strong>
          </span>
        </NavItem>
        <NavItem>
          LOGOUT
        </NavItem>
      </Fragment>
    );

    const guestLinks = (
      <Fragment>
        <NavItem>
          <Link  to={"/login"}>Login</Link>
        </NavItem>
        <NavItem>
          <Link  to={"/register"}>Register</Link>
        </NavItem>
      </Fragment>
    );

    return (
      <div>
        <Navbar color='dark' dark expand='sm' className='mb-5'>
          <Container>
            <NavbarBrand href='/'><span className="fa fa-send"></span>SendIT</NavbarBrand>
            <NavbarToggler onClick={this.toggle} />
            <Collapse navbar>
              <Nav className='ml-auto' navbar>
                {isAuthenticated ? authLinks : guestLinks}
              </Nav>
            </Collapse>
          </Container>
        </Navbar>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  auth: state.auth
});

export default connect(
  mapStateToProps,
  null
)(Navigationbar);