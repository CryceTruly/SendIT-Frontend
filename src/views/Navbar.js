import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";

const Nav = (props) => {
  const { isAuthenticated, user,onLogoutClick } = props;

  const authLinks = (
    <ul className="navbar-nav ml-auto">
      <form className="form-inline my-2 my-lg-0">
        <input className="form-control mr-sm-2" type="search" placeholder="Search" aria-label="Search" />
      </form>

      <li className="nav-item">
        <Link className="nav-link" to="/">  <strong>{user ? `Welcome ${user.username}` : ""}</strong> </Link>
      </li>
      <li className="nav-item">
        {user && user.is_admin ? (<Link className="nav-link" to="/new">All Activity</Link>) : (<Link className="nav-link" to="/new">Create Order</Link>
        )}
      </li>
      <li className="nav-item" onClick={onLogoutClick}>
        <Link className="nav-link" to="">Logout</Link>
      </li>
    </ul>
  );

  const guestLinks = (
    <ul className="navbar-nav ml-auto">
      <li className="nav-item">
        <Link className="nav-link" to="/login">Login </Link>
      </li>
      <li className="nav-item">
        <Link className="nav-link" to="/register"> Register</Link>
      </li>
    </ul>
  );
  return (

    <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
      <div className="container">
        <span className="navbar-brand"><i class="fa fa-paper-plane" aria-hidden="true"></i>SendIT</span>
        <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
          <span className="navbar-toggler-icon" />
        </button>

        <div className="collapse navbar-collapse" id="navbarSupportedContent">
          {isAuthenticated ? authLinks : guestLinks}
        </div>
      </div>
    </nav>
  );
};

export default Nav;
