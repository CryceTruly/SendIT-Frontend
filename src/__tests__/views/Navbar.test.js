import React from "react";
import { shallow } from "enzyme";
import Navbar from "../../views/Navbar";

const props = {
  isAuthenticated: true,
  user: {

  },
  onLogoutClick: jest.fn(),
};

describe("Navbar View Tests", () => {
  const wrapper = shallow(<Navbar {...props} />);
  it("should not regress", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
