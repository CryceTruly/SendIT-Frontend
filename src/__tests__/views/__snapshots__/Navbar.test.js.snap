// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar View Tests should not regress 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav
    isAuthenticated={true}
    onLogoutClick={[MockFunction]}
    user={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
      >
        <span
          className="navbar-brand"
        >
          <i
            aria-hidden="true"
            class="fa fa-paper-plane"
          />
          SendIT
        </span>
        <button
          aria-controls="navbarSupportedContent"
          aria-expanded="false"
          aria-label="Toggle navigation"
          className="navbar-toggler"
          data-target="#navbarSupportedContent"
          data-toggle="collapse"
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          id="navbarSupportedContent"
        >
          <ul
            className="navbar-nav ml-auto"
          >
            <form
              className="form-inline my-2 my-lg-0"
            >
              <input
                aria-label="Search"
                className="form-control mr-sm-2"
                placeholder="Search"
                type="search"
              />
            </form>
            <li
              className="nav-item"
            >
              <Link
                className="nav-link"
                replace={false}
                to="/"
              >
                  
                <strong>
                  Welcome undefined
                </strong>
                 
              </Link>
            </li>
            <li
              className="nav-item"
            >
              <Link
                className="nav-link"
                replace={false}
                to="/new"
              >
                Create Order
              </Link>
            </li>
            <li
              className="nav-item"
              onClick={[MockFunction]}
            >
              <Link
                className="nav-link"
                replace={false}
                to=""
              >
                Logout
              </Link>
            </li>
          </ul>
        </div>
      </div>,
      "className": "navbar navbar-expand-lg navbar-light bg-light",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="navbar-brand"
          >
            <i
              aria-hidden="true"
              class="fa fa-paper-plane"
            />
            SendIT
          </span>,
          <button
            aria-controls="navbarSupportedContent"
            aria-expanded="false"
            aria-label="Toggle navigation"
            className="navbar-toggler"
            data-target="#navbarSupportedContent"
            data-toggle="collapse"
            type="button"
          >
            <span
              className="navbar-toggler-icon"
            />
          </button>,
          <div
            className="collapse navbar-collapse"
            id="navbarSupportedContent"
          >
            <ul
              className="navbar-nav ml-auto"
            >
              <form
                className="form-inline my-2 my-lg-0"
              >
                <input
                  aria-label="Search"
                  className="form-control mr-sm-2"
                  placeholder="Search"
                  type="search"
                />
              </form>
              <li
                className="nav-item"
              >
                <Link
                  className="nav-link"
                  replace={false}
                  to="/"
                >
                    
                  <strong>
                    Welcome undefined
                  </strong>
                   
                </Link>
              </li>
              <li
                className="nav-item"
              >
                <Link
                  className="nav-link"
                  replace={false}
                  to="/new"
                >
                  Create Order
                </Link>
              </li>
              <li
                className="nav-item"
                onClick={[MockFunction]}
              >
                <Link
                  className="nav-link"
                  replace={false}
                  to=""
                >
                  Logout
                </Link>
              </li>
            </ul>
          </div>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                aria-hidden="true"
                class="fa fa-paper-plane"
              />,
              "SendIT",
            ],
            "className": "navbar-brand",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-hidden": "true",
                "class": "fa fa-paper-plane",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "SendIT",
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-controls": "navbarSupportedContent",
            "aria-expanded": "false",
            "aria-label": "Toggle navigation",
            "children": <span
              className="navbar-toggler-icon"
            />,
            "className": "navbar-toggler",
            "data-target": "#navbarSupportedContent",
            "data-toggle": "collapse",
            "type": "button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "navbar-toggler-icon",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ul
              className="navbar-nav ml-auto"
            >
              <form
                className="form-inline my-2 my-lg-0"
              >
                <input
                  aria-label="Search"
                  className="form-control mr-sm-2"
                  placeholder="Search"
                  type="search"
                />
              </form>
              <li
                className="nav-item"
              >
                <Link
                  className="nav-link"
                  replace={false}
                  to="/"
                >
                    
                  <strong>
                    Welcome undefined
                  </strong>
                   
                </Link>
              </li>
              <li
                className="nav-item"
              >
                <Link
                  className="nav-link"
                  replace={false}
                  to="/new"
                >
                  Create Order
                </Link>
              </li>
              <li
                className="nav-item"
                onClick={[MockFunction]}
              >
                <Link
                  className="nav-link"
                  replace={false}
                  to=""
                >
                  Logout
                </Link>
              </li>
            </ul>,
            "className": "collapse navbar-collapse",
            "id": "navbarSupportedContent",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <form
                  className="form-inline my-2 my-lg-0"
                >
                  <input
                    aria-label="Search"
                    className="form-control mr-sm-2"
                    placeholder="Search"
                    type="search"
                  />
                </form>,
                <li
                  className="nav-item"
                >
                  <Link
                    className="nav-link"
                    replace={false}
                    to="/"
                  >
                      
                    <strong>
                      Welcome undefined
                    </strong>
                     
                  </Link>
                </li>,
                <li
                  className="nav-item"
                >
                  <Link
                    className="nav-link"
                    replace={false}
                    to="/new"
                  >
                    Create Order
                  </Link>
                </li>,
                <li
                  className="nav-item"
                  onClick={[MockFunction]}
                >
                  <Link
                    className="nav-link"
                    replace={false}
                    to=""
                  >
                    Logout
                  </Link>
                </li>,
              ],
              "className": "navbar-nav ml-auto",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <input
                    aria-label="Search"
                    className="form-control mr-sm-2"
                    placeholder="Search"
                    type="search"
                  />,
                  "className": "form-inline my-2 my-lg-0",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-label": "Search",
                    "className": "form-control mr-sm-2",
                    "placeholder": "Search",
                    "type": "search",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                "type": "form",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    className="nav-link"
                    replace={false}
                    to="/"
                  >
                      
                    <strong>
                      Welcome undefined
                    </strong>
                     
                  </Link>,
                  "className": "nav-item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      "  ",
                      <strong>
                        Welcome undefined
                      </strong>,
                      " ",
                    ],
                    "className": "nav-link",
                    "replace": false,
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": Array [
                    "  ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Welcome undefined",
                      },
                      "ref": null,
                      "rendered": "Welcome undefined",
                      "type": "strong",
                    },
                    " ",
                  ],
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    className="nav-link"
                    replace={false}
                    to="/new"
                  >
                    Create Order
                  </Link>,
                  "className": "nav-item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Create Order",
                    "className": "nav-link",
                    "replace": false,
                    "to": "/new",
                  },
                  "ref": null,
                  "rendered": "Create Order",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    className="nav-link"
                    replace={false}
                    to=""
                  >
                    Logout
                  </Link>,
                  "className": "nav-item",
                  "onClick": [MockFunction],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Logout",
                    "className": "nav-link",
                    "replace": false,
                    "to": "",
                  },
                  "ref": null,
                  "rendered": "Logout",
                  "type": [Function],
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <span
            className="navbar-brand"
          >
            <i
              aria-hidden="true"
              class="fa fa-paper-plane"
            />
            SendIT
          </span>
          <button
            aria-controls="navbarSupportedContent"
            aria-expanded="false"
            aria-label="Toggle navigation"
            className="navbar-toggler"
            data-target="#navbarSupportedContent"
            data-toggle="collapse"
            type="button"
          >
            <span
              className="navbar-toggler-icon"
            />
          </button>
          <div
            className="collapse navbar-collapse"
            id="navbarSupportedContent"
          >
            <ul
              className="navbar-nav ml-auto"
            >
              <form
                className="form-inline my-2 my-lg-0"
              >
                <input
                  aria-label="Search"
                  className="form-control mr-sm-2"
                  placeholder="Search"
                  type="search"
                />
              </form>
              <li
                className="nav-item"
              >
                <Link
                  className="nav-link"
                  replace={false}
                  to="/"
                >
                    
                  <strong>
                    Welcome undefined
                  </strong>
                   
                </Link>
              </li>
              <li
                className="nav-item"
              >
                <Link
                  className="nav-link"
                  replace={false}
                  to="/new"
                >
                  Create Order
                </Link>
              </li>
              <li
                className="nav-item"
                onClick={[MockFunction]}
              >
                <Link
                  className="nav-link"
                  replace={false}
                  to=""
                >
                  Logout
                </Link>
              </li>
            </ul>
          </div>
        </div>,
        "className": "navbar navbar-expand-lg navbar-light bg-light",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="navbar-brand"
            >
              <i
                aria-hidden="true"
                class="fa fa-paper-plane"
              />
              SendIT
            </span>,
            <button
              aria-controls="navbarSupportedContent"
              aria-expanded="false"
              aria-label="Toggle navigation"
              className="navbar-toggler"
              data-target="#navbarSupportedContent"
              data-toggle="collapse"
              type="button"
            >
              <span
                className="navbar-toggler-icon"
              />
            </button>,
            <div
              className="collapse navbar-collapse"
              id="navbarSupportedContent"
            >
              <ul
                className="navbar-nav ml-auto"
              >
                <form
                  className="form-inline my-2 my-lg-0"
                >
                  <input
                    aria-label="Search"
                    className="form-control mr-sm-2"
                    placeholder="Search"
                    type="search"
                  />
                </form>
                <li
                  className="nav-item"
                >
                  <Link
                    className="nav-link"
                    replace={false}
                    to="/"
                  >
                      
                    <strong>
                      Welcome undefined
                    </strong>
                     
                  </Link>
                </li>
                <li
                  className="nav-item"
                >
                  <Link
                    className="nav-link"
                    replace={false}
                    to="/new"
                  >
                    Create Order
                  </Link>
                </li>
                <li
                  className="nav-item"
                  onClick={[MockFunction]}
                >
                  <Link
                    className="nav-link"
                    replace={false}
                    to=""
                  >
                    Logout
                  </Link>
                </li>
              </ul>
            </div>,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  aria-hidden="true"
                  class="fa fa-paper-plane"
                />,
                "SendIT",
              ],
              "className": "navbar-brand",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-hidden": "true",
                  "class": "fa fa-paper-plane",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "SendIT",
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-controls": "navbarSupportedContent",
              "aria-expanded": "false",
              "aria-label": "Toggle navigation",
              "children": <span
                className="navbar-toggler-icon"
              />,
              "className": "navbar-toggler",
              "data-target": "#navbarSupportedContent",
              "data-toggle": "collapse",
              "type": "button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "navbar-toggler-icon",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ul
                className="navbar-nav ml-auto"
              >
                <form
                  className="form-inline my-2 my-lg-0"
                >
                  <input
                    aria-label="Search"
                    className="form-control mr-sm-2"
                    placeholder="Search"
                    type="search"
                  />
                </form>
                <li
                  className="nav-item"
                >
                  <Link
                    className="nav-link"
                    replace={false}
                    to="/"
                  >
                      
                    <strong>
                      Welcome undefined
                    </strong>
                     
                  </Link>
                </li>
                <li
                  className="nav-item"
                >
                  <Link
                    className="nav-link"
                    replace={false}
                    to="/new"
                  >
                    Create Order
                  </Link>
                </li>
                <li
                  className="nav-item"
                  onClick={[MockFunction]}
                >
                  <Link
                    className="nav-link"
                    replace={false}
                    to=""
                  >
                    Logout
                  </Link>
                </li>
              </ul>,
              "className": "collapse navbar-collapse",
              "id": "navbarSupportedContent",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <form
                    className="form-inline my-2 my-lg-0"
                  >
                    <input
                      aria-label="Search"
                      className="form-control mr-sm-2"
                      placeholder="Search"
                      type="search"
                    />
                  </form>,
                  <li
                    className="nav-item"
                  >
                    <Link
                      className="nav-link"
                      replace={false}
                      to="/"
                    >
                        
                      <strong>
                        Welcome undefined
                      </strong>
                       
                    </Link>
                  </li>,
                  <li
                    className="nav-item"
                  >
                    <Link
                      className="nav-link"
                      replace={false}
                      to="/new"
                    >
                      Create Order
                    </Link>
                  </li>,
                  <li
                    className="nav-item"
                    onClick={[MockFunction]}
                  >
                    <Link
                      className="nav-link"
                      replace={false}
                      to=""
                    >
                      Logout
                    </Link>
                  </li>,
                ],
                "className": "navbar-nav ml-auto",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <input
                      aria-label="Search"
                      className="form-control mr-sm-2"
                      placeholder="Search"
                      type="search"
                    />,
                    "className": "form-inline my-2 my-lg-0",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "aria-label": "Search",
                      "className": "form-control mr-sm-2",
                      "placeholder": "Search",
                      "type": "search",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                  "type": "form",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      className="nav-link"
                      replace={false}
                      to="/"
                    >
                        
                      <strong>
                        Welcome undefined
                      </strong>
                       
                    </Link>,
                    "className": "nav-item",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": Array [
                        "  ",
                        <strong>
                          Welcome undefined
                        </strong>,
                        " ",
                      ],
                      "className": "nav-link",
                      "replace": false,
                      "to": "/",
                    },
                    "ref": null,
                    "rendered": Array [
                      "  ",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Welcome undefined",
                        },
                        "ref": null,
                        "rendered": "Welcome undefined",
                        "type": "strong",
                      },
                      " ",
                    ],
                    "type": [Function],
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      className="nav-link"
                      replace={false}
                      to="/new"
                    >
                      Create Order
                    </Link>,
                    "className": "nav-item",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Create Order",
                      "className": "nav-link",
                      "replace": false,
                      "to": "/new",
                    },
                    "ref": null,
                    "rendered": "Create Order",
                    "type": [Function],
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      className="nav-link"
                      replace={false}
                      to=""
                    >
                      Logout
                    </Link>,
                    "className": "nav-item",
                    "onClick": [MockFunction],
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Logout",
                      "className": "nav-link",
                      "replace": false,
                      "to": "",
                    },
                    "ref": null,
                    "rendered": "Logout",
                    "type": [Function],
                  },
                  "type": "li",
                },
              ],
              "type": "ul",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
